specs/directory.md

```diff
@@ -0,0 +1,395 @@
+| `CONFIG`  | `$XDG_CONFIG_HOME` | `%AppData%`      | `~/Library/Application Support` |
+| `CACHE`   | `$XDG_CACHE_HOME`  | `%LocalAppData%` | `~/Library/Caches`              |
+
+On Unix, `$XDG_CONFIG_HOME` is default to `~/.config` and `$XDG_CACHE_HOME` is default to `~/.cache` if XDG environment variables are empty.
+
+There is no default `BIN` path at user level since the `notation` binary can be put anywhere as long as it in the `PATH` environment variable. Common directories on Unix/Darwin are `~/bin` and `~/.local/bin` where manual `PATH` update by users may be required.
+
+## Structure
+
+The overall directory structure for `notation` is summarized as follows.
+
+```
+{BIN}
+└── notation
+{CACHE}
+└── notation
+    └── signatures
+        └── {manifest-digest-algorithm}
+            └── {manifest-digest}
+                └── {signature-digest-algorithm}
+                    └── {signature-digest}.sig
+{CONFIG}
+└── notation
+    ├── config.json
+    ├── signing
+    │   ├── certificates
+    │   │   └── {key-name}.crt
+    │   ├── keys
+    │   │   └── {key-name}.pem
+    │   └── keys.json
+    └── trust
+        ├── policy.json
+        └── store
+            └── {trust-store-type}
+                └── {named-store}
+                    └── {cert-file}
+{LIBEXEC}
+└── notation
+    └── plugins
+        └── {plugin-name}
+            └── notation-{plugin-name}
+```
+
+### Notation Binary
+
+The path for the `notation` binary is
+
+```
+{BIN}/notation
+```
+
+On Windows, the `.exe` extension is required for executables.
+
+```
+{BIN}/notation.exe
+```
+
+### Plugin
+
+[Plugins][Plugin] are binaries not meant to be executed directly by users' shell or scripts. The path of a plugin follows the pattern below.
+
+```
+{LIBEXEC}/notation/plugins/{plugin-name}/notation-{plugin-name}
+```
+
+On Windows, the `.exe` extension is required for executables.
+
+```
+{LIBEXEC}/notation/plugins/{plugin-name}/notation-{plugin-name}.exe
+```
+
+### General Configuration
+
+The path of the general configuration file of the `notation` CLI is
+
+```
+{CONFIG}/notation/config.json
+```
+
+### Trust Store
+
+The path of a certificate file in a [Trust Store][TS] follows the pattern of
+
+```
+{CONFIG}/notation/trust/store/{trust-store-type}/{named-store}/{cert-file}
+```
+
+### Trust Policy
+
+The path of the [Trust Policy][TP] file is
+
+```
+{CONFIG}/notation/trust/policy.json
+```
+
+### Signature Caches
+
+The signatures are cached to optimize the network traffic. The path of a cached signature for a certain manifest follows the pattern below.
+
+```
+{CACHE}/notation/signatures/{manifest-digest-algorithm}/{manifest-digest}/{signature-digest-algorithm}/{signature-digest}.sig
+```
+
+or in a hierarchical view
+
+```
+{CACHE}
+└── notation
+    └── signatures
+        └── {manifest-digest-algorithm}
+            └── {manifest-digest}
+                └── {signature-digest-algorithm}

```

Comment:
Should we rename `{signature-digest-algorithm}` to `{signature-blob-digest-algorithm}` to make it clearer, I (incorrectly) assumed this was related to signing algorithm hash earlier and that was confusing.